### PR TITLE
All Test Cases implemented with Dynamic timing and centered elements

### PR DESCRIPTION
**Pull request by skippynose2**

### CODE REVIEW

File: tests/TestGUI.py [hunk 1]
```diff
@@ -107,7 +108,7 @@ def __show_data_and_confirm(test_data_path: str):
         # Setup the window and canvas
         popup = tk.Toplevel()
         popup.wm_title("Data Confirmation")
-        popup.state('zoomed')
+        popup.state('zoomed')  # TODO unix-based systems use -zoomed
```
Comment (ian1dunn) [inline tests/TestGUI.py:111]: Please fix this so we don't have to keep changing every commit.

**Review by ian1dunn — COMMENTED** [review]
I merged from dev and there were a couple of merge conflicts that I fixed. Please do this before opening a PR in the future.

A couple of things:
- [ ] When I was testing, the program errored out on every test (probably from the merge conflicts). Please fix.
- [ ] Update docstrings for tests (they are all currently ones from Blink)
- [ ] Update changelog

File: tests/TestGUI.py (hunk 1 above)
Comment (skippynose2) [reply tests/TestGUI.py:111]: Yep will fix

**Comment by skippynose2** [comment]
> I merged from dev and there were a couple of merge conflicts that I fixed. Please do this before opening a PR in the future.
> 
> A couple of things:
> 
> * [ ]  When I was testing, the program errored out on every test (probably from the merge conflicts). Please fix.
> * [ ]  Update docstrings for tests (they are all currently ones from Blink)
> * [ ]  Update changelog

The program erroring out is the (_tkinter.TclError: bad window path name) issue. I am not sure why this happening but data collection and saving is still functioning as normal.

File: tests/TestGUI.py (hunk 1 above)
Comment (ian1dunn) [reply tests/TestGUI.py:111]: Fixed in branch dev-bugFixes

**Pull request closed without merging**